feature/select-subject-age-group/src/main/java/com/simprints/feature/selectagegroup/screen/BuildAgeGroupsDescriptionUseCase.kt

```diff
@@ -0,0 +1,94 @@
+package com.simprints.feature.selectagegroup.screen
+
+import android.content.Context
+import com.simprints.infra.config.store.ConfigRepository
+import com.simprints.infra.config.store.models.AgeGroup
+import com.simprints.infra.config.store.models.allowedAgeRanges
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import com.simprints.infra.resources.R.string as IDR
+
+internal class BuildAgeGroupsDescriptionUseCase @Inject constructor(
+    private val configurationRepo: ConfigRepository,
+    @ApplicationContext private val context: Context,
+) {
+
+    /**
+     * Builds a list of age groups for display
+     * it reads the allowed age ranges from the configuration and formats them for display
+     * it also adds a 0-<first age range> and <last age range>-above if they are not present
+     */
+    suspend operator fun invoke(): List<AgeGroupDisplayModel> {
+
+        val allowedAgeRanges = configurationRepo.getProjectConfiguration().allowedAgeRanges()
+        val processedAgeRanges = generateSortedUniqueAgeGroups(allowedAgeRanges)
+
+        return processedAgeRanges.map { ageGroup ->
+            AgeGroupDisplayModel(ageGroup.getDisplayName(), ageGroup)
+        }
+    }
+
+    private fun generateSortedUniqueAgeGroups(ageGroups: List<AgeGroup>): List<AgeGroup> {
+        // Handle empty list case by returning a single age group starting at 0 and ending with null
+        if (ageGroups.isEmpty()) return listOf(AgeGroup(0, null))
```

Comment:
This should not happen as you are only displaying the age selection if there are some age groups no?